app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreen.kt

```diff
@@ -3,30 +3,94 @@ package org.simple.clinic.bloodsugar.history
 import android.content.Context
 import android.util.AttributeSet
 import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import kotlinx.android.synthetic.main.screen_blood_sugar_history.view.*
+import org.simple.clinic.R
 import org.simple.clinic.bloodsugar.BloodSugarMeasurement
+import org.simple.clinic.bloodsugar.history.adapter.BloodSugarHistoryListItem
+import org.simple.clinic.bloodsugar.history.adapter.BloodSugarHistoryListItemDiffCallback
 import org.simple.clinic.di.injector
+import org.simple.clinic.patient.DateOfBirth
+import org.simple.clinic.patient.Gender
 import org.simple.clinic.patient.Patient
+import org.simple.clinic.patient.displayLetterRes
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.widgets.DividerItemDecorator
+import org.simple.clinic.widgets.ItemAdapter
+import org.simple.clinic.widgets.dp
+import org.threeten.bp.format.DateTimeFormatter
 import java.util.UUID
+import javax.inject.Inject
+import javax.inject.Named
 
 class BloodSugarHistoryScreen(
     context: Context,
     attrs: AttributeSet
 ) : ConstraintLayout(context, attrs), BloodSugarHistoryScreenUi, BloodSugarHistoryScreenUiActions {
 
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  @Inject
+  lateinit var userClock: UserClock
+
+  @field:[Inject Named("date_for_bp_history")]
```

Comment:
The name indicates this is only for `bp` history while this is being used in the blood sugar screen. Since it's the same format in both screens, maybe we can find a common name for them, say `measurement_*` instead of `bp_*`.

What do you think?

Reply:
I was hoping to discuss it later and implement it in later PR. But I will change it now 👍